.github/workflows/test.yml

```diff
@@ -82,8 +84,23 @@ jobs:
           sudo curl -L https://github.com/docker/compose/releases/download/${{ matrix.compose_version }}/docker-compose-`uname -s`-`uname -m` -o "${{ matrix.compose_path }}/docker-compose"
           sudo chmod +x "${{ matrix.compose_path }}/docker-compose"
 
+      - name: Install self-hosted
+        uses: nick-fields/retry@v3
+        with:
+          max_attempts: 3
+          timeout_minutes: 10
+          retry_on: error
+          command: |
+            export REPORT_SELF_HOSTED_ISSUES=0
+            ./install.sh
+
       - name: Integration Test
-        run: ./integration-test.sh
+        uses: nick-fields/retry@v3
+        with:
+          max_attempts: 3
```

Comment:
Just so I understand: what was the behavior before you added these settings? Unlimited retries? No timeout so it hung until the action crashed?

Reply:
Yep, that's correct. The `max_attempts` really is just the first step into adding flaky test detection. If a job fails, but then is retried and succeed, it can be marked as flaky. The `timeout_minutes` is a required parameter here. I can remove this and readd in a follow-up PR if that is more clear

Reply:
No, that's fine. I just wanted to understand the change. LGTM!